jsonrpc/codec.go

```diff
@@ -102,6 +103,78 @@ const (
 
 type BlockNumber int64
 
+type BlockNumberOrHash struct {
+	BlockNumber *BlockNumber `json:"blockNumber,omitempty"`
+	BlockHash   *types.Hash  `json:"blockHash,omitempty"`
+}
+
+// Unmarshal will try to extract the filter's data.
+// Here are the possible input formats :
+//
+// 1 - "latest", "pending" or "earliest"	- self-explaining keywords
+// 2 - "0x2"								- block number #2 (EIP-1898 backward compatible)
+// 3 - {blockNumber:	"0x2"}				- EIP-1898 compliant block number #2
+// 4 - {blockHash:		"0xe0e..."}			- EIP-1898 compliant block hash 0xe0e...
+func (bnh *BlockNumberOrHash) Unmarshal(input *interface{}) error {
+	var placeholder BlockNumberOrHash
+
+	data, err := json.Marshal(*input)
+	if err != nil {
+		return fmt.Errorf("failed to serialize input: %v", err)
+	}
+
+	err = json.Unmarshal(data, &placeholder)
+	if err != nil {
+		var keyword string
+		err = json.Unmarshal(data, &keyword)
```

Comment:
Aren't you repeating here what it is already done in ```stringToBlockNumber```?

Reply:
Good catch 👍 Updated in https://github.com/0xPolygon/polygon-sdk/pull/284/commits/0d3704c7e71429ee06a365926544f6018716d771